src/core/thread/mle_router.cpp

```diff
@@ -2646,15 +2646,22 @@ void MleRouter::HandleChildUpdateRequest(RxInfo &aRxInfo)
         CslChannelTlv cslChannel;
         uint32_t      cslTimeout;
 
-        if (Tlv::Find<CslTimeoutTlv>(aRxInfo.mMessage, cslTimeout) == kErrorNone)
+        switch (Tlv::Find<CslTimeoutTlv>(aRxInfo.mMessage, cslTimeout))
```

Comment:
No need here. It is already checked by this flavor of `Find<Tlv>` method. It can return `kErrorParse` if it is not valid (and `kErrorNotFound` if not found and `kErrorNone` if successfully parsed).

Unlike `FindTlv` (which treats the TLV as opaque set of bytes and cannot check its validity) this flavor of `Find<Tlv>` knows about the expected value (e.g. here we pass a  `uint32_t cslTimeout`), and makes sure we can read it. It also does  compile-time check for the value type per TLV(e.g., if we pass a `uint16_t` for `CslTimeoutTlv` we get a build-time error saying this TLV expects `uint32_t`).

Reply:
Thanks, good explanation.